src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -169,24 +176,23 @@ void setupParser(InputStream queryResponseStream) throws IOException {
    * @throws AsyncApiException  if there is an issue creating the job
    * @throws InterruptedException sleep interrupted
    */
-  public InputStream waitForBatchResults(BulkConnection bulkConnection, String jobId, String batchId)
+  public String[] waitForBatchResults(BulkConnection bulkConnection, String jobId, String batchId)
     throws AsyncApiException, InterruptedException {
 
     BatchInfo info = null;
     for (int i = 0; i < SalesforceSourceConstants.GET_BATCH_RESULTS_TRIES; i++) {
-      info = bulkConnection.getBatchInfo(jobId, batchId);
-
-      if (info.getState() == BatchStateEnum.Completed) {
-        QueryResultList list =
-          bulkConnection.getQueryResultList(jobId, batchId);
-        String[] resultIds = list.getResult();
-
-        List<InputStream> streams = new ArrayList<>(resultIds.length);
-        for (String resultId : resultIds) {
-          streams.add(bulkConnection.getQueryResultStream(jobId, batchId, resultId));
+      try {
+        info = bulkConnection.getBatchInfo(jobId, batchId);
+      } catch (AsyncApiException e) {
+        if (i == SalesforceSourceConstants.GET_BATCH_RESULTS_TRIES - 1) {
+          throw e;
         }
+        LOG.warn("Failed to get batch info: {}, retry {}", e, i);
```

Comment:
exceptions should be the last thing in the log. Something like:

`LOG.warn("Failed to get info for batch {}. Will retry after some time.", batchId, e);`

Reply:
fixed